### PR TITLE
ratelimit: Change rate limit service type to cluster

### PR DESCRIPTION
**Pull request by cnvergence**
**Which issue(s) this PR fixes**:

Fixes 
- https://github.com/envoyproxy/gateway/issues/1577

### CODE REVIEW

File: internal/infrastructure/kubernetes/ratelimit/resource_provider.go [hunk 1]
```diff
@@ -54,8 +54,11 @@ func (r *ResourceRender) Service() (*corev1.Service, error) {
 	}
 
 	labels := rateLimitLabels()
+	if r.rateLimitDeployment.ServiceType == nil {
+		r.rateLimitDeployment.ServiceType = egcfgv1a1.DefaultKubernetesServiceType()
+	}
 
-	serviceSpec := resource.ExpectedServiceSpec(egcfgv1a1.DefaultKubernetesServiceType())
+	serviceSpec := resource.ExpectedServiceSpec(egcfgv1a1.GetKubernetesServiceType(*r.rateLimitDeployment.ServiceType))
```
Comment (arkodg) [inline internal/infrastructure/kubernetes/ratelimit/resource_provider.go:61]: I dont think we need to define an API here, should be okay to hardcode to `Cluster`
the Rate Limit Service is internal to the Envoys living in the same cluster (and same namespace), so imo there is no need
to expose it to clients outside the cluster using `NodePort` or `LoadBalancer`

Comment (cnvergence) [reply internal/infrastructure/kubernetes/ratelimit/resource_provider.go]: This makes sense to me, I will write in the original issue that there will be just a change of the default setting.

**Review by arkodg — CHANGES_REQUESTED** [review]
this doesn't look right, we dont want to change the default API value, which is used for the Envoy (proxy) service, which should default to LoadBalancer.
we need to somehow change the service type within the rate limit infra code to always be set to `Cluster`

**Review by arkodg — APPROVED** [review]
LGTM thanks, would be good if you could also add a comment highlighting the "why" - its internal to the envoys running in the cluster

**Review by zirain — APPROVED** [review]

**Pull request merged**